sstables/sstable_directory.cc

```diff
@@ -571,22 +571,22 @@ bool sstable_directory::compare_sstable_storage_prefix(const sstring& prefix_a,
     return size_a == size_b && sstring::traits_type::compare(prefix_a.begin(), prefix_b.begin(), size_a) == 0;
 }
 
-future<std::unordered_map<sstring, sstring>> sstable_directory::create_pending_deletion_log(const std::vector<shared_sstable>& ssts) {
-    return seastar::async([&ssts] {
-        std::unordered_map<sstring, min_max_tracker<generation_type>> gen_trackers;
-        std::unordered_map<sstring, sstring> res;
+future<sstable_directory::pending_delete_result> sstable_directory::create_pending_deletion_log(opened_directory& base_dir, const std::vector<shared_sstable>& ssts) {
```

Comment:
We should encapsulate this in a class for better readability.

Reply:
encapsulate what?  the pending_delete_log functions and data structures?